backend/Application/Aggregates/Contract/Entities/ContractSnapshot.cs

```diff
@@ -0,0 +1,198 @@
+using System.Threading.Tasks;
+using Application.Aggregates.Contract.Types;
+using Application.Api.GraphQL;
+using Application.Api.GraphQL.Transactions;
+using HotChocolate;
+
+namespace Application.Aggregates.Contract.Entities;
+
+/// <summary>
+/// Contains aggregated local for <see cref="Contract"/>.
+/// </summary>
+public sealed class ContractSnapshot
+{
+    public ulong BlockHeight { get; init; }
+    public ulong ContractAddressIndex { get; init; }
+    public ulong ContractAddressSubIndex { get; init; }
+    public string ContractName { get; init; }
+    public string ModuleReference { get; init; }
+    public double Amount { get; init; }
+    
+    [GraphQLIgnore]
+    public ImportSource Source { get; init; }
+    [GraphQLIgnore]
+    public DateTimeOffset CreatedAt { get; init; } = DateTime.UtcNow;
+
+    /// <summary>
+    /// Needed for EF Core
+    /// </summary>
+    private ContractSnapshot()
+    {
+    }
+
+    internal ContractSnapshot(
+        ulong blockHeight,
+        ContractAddress contractAddress,
+        string contractName,
+        string moduleReference,
+        double amount,
+        ImportSource importSource
+        )
+    {
+        BlockHeight = blockHeight;
+        ContractAddressIndex = contractAddress.Index;
+        ContractAddressSubIndex = contractAddress.SubIndex;
+        ContractName = contractName;
+        ModuleReference = moduleReference;
+        Amount = amount;
+        Source = importSource;
+    }
+
+    internal static async Task ImportContractSnapshot(
+        IContractRepositoryFactory repositoryFactory,
+        IContractRepository repository, 
+        ImportSource source
+        )
+    {
+        await using var moduleReadonlyRepository = await repositoryFactory.CreateModuleReadonlyRepository();
+            
+        var contractEventsAdded = repository
+            .GetContractEventsAddedInTransaction()
+            .ToList();
+        
+        var addressesInitialized = contractEventsAdded
+            .Where(ce => ce.Event is ContractInitialized)
+            .Select(ce => (ce.ContractAddressIndex, ce.ContractAddressSubIndex))
+            .ToHashSet();
+
+        var addressesUpdated = contractEventsAdded
+            .Where(ce => !addressesInitialized.Contains((ce.ContractAddressIndex, ce.ContractAddressSubIndex)))
```

Comment:
Do we only care about update events for contracts that were just initialized? Maybe it is just me, who don't really understands the context this is running in.

Reply:
I’m not sure I follow your question. This comment is linked to logic, to contract which doesn’t have a initialization event in the scope of contract events given.

Could you try clarify more, what your concern is?

Reply:
Oh, I missed the `!` in the front.